staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -186,6 +185,13 @@ func ValidateCustomResourceDefinitionValidation(customResourceValidation *apiext
 	if customResourceValidation.OpenAPIV3Schema != nil {
 		openAPIV3Schema := &specStandardValidatorV3{}
 		allErrs = append(allErrs, ValidateCustomResourceDefinitionOpenAPISchema(customResourceValidation.OpenAPIV3Schema, fldPath.Child("openAPIV3Schema"), openAPIV3Schema)...)
+		
+		// if validation passed otherwise, make sure we can actually construct a schema validator from this schema
+		if len(allErrs) == 0 {
+			if _, err := apiservervalidation.NewValidator(customResourceValidation.OpenAPIV3Schema); err != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("openAPIV3Schema"), "", fmt.Sprintf("error building validator: %v", err)))
```

Comment:
It would be better to display the schema in the error, instead of having `""` i.e.

```go
field.Invalid(fldPath.Child("openAPIV3Schema"), customResourceValidation.OpenAPIV3Schema, fmt.Sprintf("error building validator: %v", err)))
```

That's also the convention followed for other `field.Invalid` errors.

Reply:
that's true, but here that would print the entire schema in struct form... it's pretty unusable

Reply:
👍

Reply:
good